### PR TITLE
chore: reformat multi-line statements

### PR DESCRIPTION
**Pull request by mwtoews**
This PR refactors some multi-line statements. This was mostly automated by temporarily toggling this control in pyproject.toml:
```toml
[tool.ruff.format]
skip-magic-trailing-comma = true
```
then selecting some commits that condense multi-line statements to single, while rejecting other statements that are better represented over multiple lines. There was no exact rule-of-thumb for the process, except to try and maintain the same style in surrounding code blocks. Class constructors or instantiations of some classes with many parameters are kept as multi-line.

Note that while there are over 2300 fewer lines with this PR, there are no functional changes.

### CODE REVIEW

**Review by wpbonelli — APPROVED** [review]

File: flopy/modflow/mfsfr2.py [hunk 1]
```diff
@@ -928,7 +923,7 @@ def load(cls, f, model, nper=None, gwt=False, nsol=1, ext_unit_dict=None):
                         # dataset 6d description suggests that this line isn't read for
                         # isfropt > 1 but description of icalc suggest that icalc=2
                         # (8-point channel) can be used with any isfropt
-                        if i == 0 or nstrm > 0 and not reachinput or isfropt <= 1:
+                        if i == 0 or (nstrm > 0 and not reachinput) or isfropt <= 1:
```
Comment (mwtoews) [inline flopy/modflow/mfsfr2.py:926]: FYI, this one was an autofix for [RUF021](https://docs.astral.sh/ruff/rules/parenthesize-chained-operators/)

File: flopy/modpath/mp7.py [hunk 2]
```diff
@@ -137,20 +137,14 @@ def __init__(
...
                     shape = (nlay, nrow, ncol)
                 elif dis.package_name.lower() == "disv":
                     nlay, ncpl = dis.nlay.array, dis.ncpl.array
                     shape = (nlay, ncpl)
                 elif dis.package_name.lower() == "disu":
                     nodes = dis.nodes.array
-                    shape = tuple(
-                        nodes,
-                    )
+                    shape = tuple(nodes)
```
Comment (mwtoews) [inline flopy/modpath/mp7.py:147]: This should probably be corrected to `shape = (nodes,)` (it was probably originally `shape = tuple(nodes,)` which is not correct)

File: flopy/utils/datafile.py [hunk 3]
```diff
@@ -221,9 +221,7 @@ def __init__(self, filename: Union[str, os.PathLike], precision, verbose, **kwar
         if self.mg is None:
             self.mg = StructuredGrid(
                 delc=np.ones((self.nrow,)),
-                delr=np.ones(
-                    self.ncol,
-                ),
+                delr=np.ones(self.ncol),
```
Comment (mwtoews) [inline flopy/utils/datafile.py:224]: This is where the code was probably originally `np.ones(self.ncol,)` in attempt to specify a tuple (which would be done with `np.ones((self.ncol,))`). Fortunately, this is harmless, as numpy's shape can be an integer to mean the same as a tuple of one integer. No change is recommended here.

File: autotest/test_grid.py [hunk 4]
```diff
@@ -379,12 +379,8 @@ def test_unstructured_xyz_intersect(example_data_path):
...
     nnodes = np.sum(ncpl)
 
-    top = np.ones(
-        (nnodes),
-    )
-    botm = np.ones(
-        (nnodes),
-    )
+    top = np.ones((nnodes))
+    botm = np.ones((nnodes))
```
Comment (mwtoews) [inline autotest/test_grid.py:383]: This probably should be corrected to `np.ones((nnodes,))` (or simply `np.ones(nnodes)`)

**Review by mwtoews — COMMENTED** [review]
Triple-check of changes complete, with a few notes below. This PR highlighted a few places where a trailing `,` was originally used, sometimes to try and construct a `tuple` type. I'll follow this up with a possible correction and amend `.git-blame-ignore-revs`.

**Pull request merged**